test/test_nn.py

```diff
@@ -10116,6 +10115,145 @@ def test_grid_sample_3d_alert_nondeterministic(self, device):
         output = F.grid_sample(input, grid, align_corners=False)
         output.sum().backward()
 
+    @dtypes(torch.float, torch.double)
+    @largeTensorTest(lambda self, device, dtype:
+                     32769 * (65536 + 3 * 65536 / 128) *
+                     torch.tensor([], dtype=dtype).element_size())
+    def test_grid_sample_large_index_2d(self, device, dtype):
```

Comment:
Because we do have a fallback implementation for `grid_sample2d` on CPU, even though I know the code was copy-pasted from a previous implementation, I would like to test it more rigorously to ensure its correctness and that there is parity between it and our existing implementations. Otherwise, it would be very easy for someone to only update the vectorized implementation and forget to add a feature or a bug fix to the fallback path.

I'm wondering if it is possible to register a new operator, `_grid_sample_2d_cpu_fallback_debug` and run the grid_sampler tests on it https://github.com/pytorch/pytorch/blob/a01e91e6b2ab28b000d66588b9d32a91f8fc4e62/test/test_nn.py#L6584 to ensure that the fallback works with small inputs. The inputs from the test doesn't necessarily test that the fallback can handle 64-bit inputs, but it'll give us confidence that the implementation is correct

Reply:
Done

>It would also be nice to do this for the autograd tests, but I am not sure where those are

`gradcheck` and `.grad` comparison is also covered in `test_grid_sample`.